lib/pure/httpclient.nim

```diff
@@ -1023,11 +1032,17 @@ proc request*(client: HttpClient | AsyncHttpClient, url: string,
   else:
     await newConnection(client, connectionUrl)
 
-  if not client.headers.hasKey("user-agent") and client.userAgent != "":
-    client.headers["User-Agent"] = client.userAgent
+  var effectiveHeaders: HttpHeaders
+  if overrideHeaders != nil:
+    effectiveHeaders = overrideHeaders.withFallback(client.headers)
```

Comment:
Let's put the nil check into the ``withFallback`` function and maybe reverse it. Call it ``override`` and change it so that it's called like so:

```
client.headers.override(overrideHeaders)
```